File: s3GoSDK/sample/PolicySerialTesting.go
package main

import (
	"context"
	"fmt"

	"github.com/aws/aws-sdk-go-v2/aws"
	"github.com/aws/aws-sdk-go-v2/service/s3"
)

func PolicySerialTest(buckets [3]string) {
	input := &s3.PutBucketPolicyInput{
		Bucket: &buckets[0],
		Policy: aws.String("{\"Version\": \"2012-10-17\", \"Statement\": [{ \"Sid\": \"DenyPublicREAD\",\"Effect\": \"Deny\",\"Principal\": {\"AWS\": \"*\"}, \"Action\": \"s3:GetObject\", \"Resource\": [\"arn:aws:s3:::" + buckets[0] + "/*\" ] } ]}"),
	}

	_, putbucketpolicyErr := Client.PutBucketPolicy(context.TODO(), input)
	if putbucketpolicyErr != nil {
		fmt.Println("Got an error PutPolicy item:")
		fmt.Println(putbucketpolicyErr)
		return
	}
	output := &s3.GetBucketPolicyInput{
		Bucket: &buckets[0],
	}
	result, getbucketpolicyErr := Client.GetBucketPolicy(context.TODO(), output)
	if getbucketpolicyErr != nil {
		fmt.Println("Got an error GetPolicy item:")
		fmt.Println(getbucketpolicyErr)
		return
	}
	fmt.Println("Get " + buckets[0] + " Policy: \n" + *result.Policy)
	fmt.Println("")
}

File: s3RustSDK/sample/src/bin/policy_serial_testing.rs
use std::process;

mod client;

use structopt::StructOpt;
use tracing_subscriber::fmt::format::FmtSpan;
use tracing_subscriber::fmt::SubscriberBuilder;
#[derive(Debug, StructOpt)]
struct Opt {
    /// The bucket name
    #[structopt(short, long)]
    bucket: String,

    /// Whether to display additional information
    #[structopt(short, long)]
    verbose: bool,
}

#[tokio::main]
async fn main() {
    let Opt {
        bucket,
        verbose,
    } = Opt::from_args();

    if verbose {
        println!("S3 client version: {}", s3::PKG_VERSION);

        SubscriberBuilder::default()
            .with_env_filter("info")
            .with_span_events(FmtSpan::CLOSE)
            .init();
    }

    match client::client()
        .put_bucket_policy()
        .policy("{\"Version\": \"2012-10-17\", \"Statement\": [{ \"Sid\": \"DenyPublicREAD\",\"Effect\": \"Deny\",\"Principal\": {\"AWS\": \"*\"}, \"Action\": \"s3:GetObject\", \"Resource\": [\"arn:aws:s3:::yuyuman1/*\" ] } ]}")
        .bucket(&bucket)
        .send()
        .await
    {
        Ok(_) => {
            println!("Created bucket {} policy", bucket);
        }    
        Err(e) => {
            println!("Got an error creating bucket policy:");
            println!("{}", e);
            process::exit(1);
        }
    };
}
